### PR TITLE
Convert CommonJS exports to ES6 exports

### PR DESCRIPTION
**Pull request by turt2live**
**This is against `travis/sourcemaps` for safety.**

Split from https://github.com/matrix-org/matrix-react-sdk/pull/3744

Part 1 of a few for ES6 conversion. **Best reviewed commit-by-commit.**

----

This PR and others in the series have their overview covered here: https://gist.github.com/turt2live/a3fc7c9712b8ef0f1f758611aa33382d

### CODE REVIEW

**Comment by turt2live** [comment]
A failing build is not surprising.

File: src/extend.js [hunk 1]
```diff
@@ -16,11 +16,11 @@ limitations under the License.
 
 'use strict';
 
-module.exports = function(dest, src) {
+export default function(dest, src) {
```
Comment (dbkr) [inline src/extend.js]: I thought `export default` broke things?

File: src/notifications/ContentRules.js [hunk 2]
```diff
@@ -16,9 +17,9 @@ limitations under the License.
 
 'use strict';
 
-const PushRuleVectorState = require('./PushRuleVectorState');
+import {PushRuleVectorState} from "./PushRuleVectorState";
 
-module.exports = {
+export class ContentRules {
```
Comment (dbkr) [inline src/notifications/ContentRules.js]: Same comment as https://github.com/matrix-org/matrix-js-sdk/pull/1134 on classes of static methods.

File: src/notifications/NotificationUtils.js [hunk 3]
```diff
@@ -16,14 +17,14 @@ limitations under the License.
 
 'use strict';
 
-module.exports = {
+export class NotificationUtils {
```
Comment (dbkr) [inline src/notifications/NotificationUtils.js]: ditto

File: src/extend.js (hunk 1 above)
Comment (turt2live) [reply src/extend.js]: It does for the js-sdk, not for the react-sdk. We don't have a central index for the react-sdk that we need to worry about

**Pull request merged**